### DISCUSSION TITLE
Anyone using Prisma Plugin exposing decimal types?

### DISCUSSION THREAD

**Discussion by marceloverdijk** [hayes/pothos#451] [Q&A] [answered]
Prisma uses decimal.js for Pisma `Decimal` type in models
I wonder how others are mapping this with Pothos.

Using e.g. `t.exposeFloat('points')` gives a typescript warning.

I'm already using [`graphql-scalars`](https://www.npmjs.com/package/graphql-scalars) for e.g. `Latitude` and `Longitude` GraphQL fields (these are also Prisma `Decimal` types).
Honestly I expected `graphql-scalar` to have a similar decimal GraphQL type but that's not the case. So I'm wondering if I'm overlooking something obvious.

**Comment by hayes** [comment, marked as answer]
I haven't used this myself.  The place to start would probably to define it as:

```ts
t.field({
  type: ['Float'], // if its a list, or just `'Float'` if its not a list
  resolve: (parent) => parent.points
})
```

This will give you a better message on why typescript thinks this is not safe.  I haven't used decimal.js myself, so not sure about how it works.  There are a few pieces at play.  When you create the builder, you can define types for the scalars in your schema (you probably already have this set up for other scalars)  This also works for built in scalars like float.  It defaults to `number` but you could include the decimal.js type there, and then typescript would likely allow that.  I don't know if the serialization logic for the GraphQL float type will work, so you would need to figure that part out on your own.  Alternatively you can convert the decminal type into a javascript number in your resolver.

Hope this helps a little, I know it's not a very complete answer

**Reply by marceloverdijk** [reply]
Yes I think I will create a custom scalar (not difficult). As I couldn't find a out-of-the-box solution I was wondering if I was doing something wrong or how others approached this; as the use case to use Decimals is quite normal I would say. Anyway thanks for your help.
I will post my solution + example later here.

**Reply by GeKVe** [reply]
Did you find a proper solution?

**Reply by hayes** [reply]
There was a related issue here with a better answer: https://github.com/hayes/pothos/issues/730

And a test case you can look at as a reference here: https://github.com/hayes/pothos/commit/5abbd9fc5df4d34a1a019296fb0a636d63d6e4bb

**Reply by GeKVe** [reply]
Thank you for your quick response. This works fine for me. 👏 ⭐
